### DISCUSSION TITLE
Why doesn't greedy_modularity_communities return the maximum modularity partitions with a weighted graph?

### DISCUSSION THREAD

**Discussion by YL-byte** [networkx/networkx#6619] [Q&A] [answered]
I ran into a problem using networkx.algorithms.community.greedy_modularity_communities. It seems that when using weights, the algorithm doesn't return the partition with the maximum modularity. Here is a small code demonstrating the problem:
```
#!/usr/bin/env python3
import networkx as nx
from networkx.algorithms.community import greedy_modularity_communities
from networkx.algorithms.community.quality import modularity

if __name__ == "__main__":
    vertices = {
        'a' : 0,
        'b' : 1,
        'c' : 2,
        'd' : 3,
        'e' : 4,
        'f' : 5
    }
    connections = [
        ['a', 'b', 1],
        ['b', 'c', 3],
        ['c', 'f', 2],
        ['f', 'e', 3],
        ['e', 'd', 2],
        ['d', 'a', 2],
        ['c', 'd', 5],
        ['b', 'f', 1],
        ['b', 'e', 4]
    ]
    graph = nx.Graph()
    for v in vertices:
        graph.add_node(v)

    for c in connections:
        graph.add_edge(c[0], c[1], weight=c[2])

    print(list(greedy_modularity_communities(graph, weight='weight'))) #Returns [frozenset({'b', 'a'}), frozenset({'c'}), frozenset({'d'}), frozenset({'e'}), frozenset({'f'})]
    print(modularity(graph, greedy_modularity_communities(graph, weight='weight'))) #Prints -0.16540642722117202
    print(modularity(graph, [frozenset({'a', 'd', 'c'}), frozenset({'b', 'e', 'f'})], weight='weight')) #Prints 0.15122873345935728
```
Is there something I'm doing wrong when running the code? It doesn't seem to make since that the result has a negative modularity especially when there is an example of a partition with a positive one. Thanks in advance!

I've tried to read the documentation and looking through old posts in several websites looking for someone with a similar problem.

Thanks!

**Comment by dschult** [comment, marked as answer]
Hmmm...   I get the results
```
[frozenset({'a', 'c', 'd'}), frozenset({'e', 'f', 'b'})]
0.15122873345935728
0.15122873345935728
```
What version of networkx are you using?   `print(nx.__version__)`

**Reply by YL-byte** [reply]
Hi,
I have version 2.4, which I now see is outdated.
I'll update and try to rerun the script.
Thanks for the quick reply!

**Reply by YL-byte** [reply]
That was indeed the problem, thanks!
